core/templates/dev/head/pages/skill_player/concept_card_directive.html

```diff
@@ -0,0 +1,78 @@
+<div class="conversation-skin-tutor-card">
+  <div class="conversation-skin-tutor-card-content">
+    <div class="conversation-skin-tutor-card-top-section">
+      <img class="conversation-skin-oppia-avatar"
+           ng-src="<[OPPIA_AVATAR_IMAGE_URL]>" alt="">
+      <div class="rte-viewer conversation-skin-tutor-card-top-content">
+        <angular-html-bind html-data="skillExplanation">
+        </angular-html-bind>
+        <div ng-if="workedExamples.length > 0" style="margin-top: 20px;">
+          <h3> Worked Examples: </h3>
+          <p ng-repeat="workedExample in workedExamples track by $index"
+             class="worked-example"
+             ng-if="$index <= maximumVisibleExampleIndex">
+            <label> <[$index + 1]>: &nbsp;</label>
+            <angular-html-bind html-data="workedExample">
+            </angular-html-bind>
+          </p>
+          <button class="btn btn-primary add-worked-example"
+                  ng-click="viewNextExample()"
+                  ng-if="maximumVisibleExampleIndex < (workedExamples.length - 1)">
+            <i class="material-icons"> add </i> Worked Example
+          </button>
+        </div>
+      </div>
+    </div>
+    <continue-button on-click-continue-button="returnToExploration()"
+                     class="oppia-learner-confirm-button">
+    </continue-button>
+  </div>
+</div>
+
+<style>
+  concept-card .oppia-learner-confirm-button {
+    margin-left: 10px;
+    margin-bottom: 10px;
+  }
+
+  concept-card .add-worked-example {
+    background: #0D48A1;
+    color: #ffffff;
+    margin-top: 2vh;
+  }
+
+  .conversation-skin-tutor-card {
```

Comment:
Is this copied from elsewhere? If so don't do that. Pull common classes out into oppia.css and give them appropriate general names that start with `.oppia-`.

Reply:
They were taken from tutor-card-directive. I did try changing names, but some classes had their style definitions split between tutor card directive and conversation skin directive. Also, all directives in the latter, like tutor card, supplementary card, progress nav etc. used these classes itself, so I thought concept card could also use it.
Should I pull out the classes from conversation skin directive into oppia,css?

Reply:
I have no objection to reusing classes, but pull out the reused classes into oppia.css and rename them to something more encompassing like .oppia-learner-view-card (say). Also, if style definitions are split, group them into one class -- that shouldn't happen.

Reply:
Done, moved some common classes to oppia.css.